drivers/clock_control/clock_control_nrf2_hsfll.c

```diff
@@ -221,7 +208,6 @@ static DEVICE_API(nrf_clock_control, hsfll_drv_api) = {
 	.std_api = {
 		.on = api_nosys_on_off,
 		.off = api_nosys_on_off,
-		.get_rate = api_get_rate_hsfll,
```

Comment:
Its not useful when the frequency is dynamic, it can change at any time, the only way to guarantee any frequency is to request what you need, and wait for confirmation, so there is no value in checking the freq like this at any time. Its also not possible with the global hsfll (not this clock I know) but it proves the point :) there we can't even get the freq if we wanted :)

Reply:
Agreed. It could be mentioned in the commit message so that the motivation is clear.